src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/BreakpointComplications.java

```diff
@@ -100,56 +145,152 @@ boolean isDupAnnotIsFromOptimization() {
+        final int jumpStartRefLoc = firstAlignmentInterval.forwardStrand ? firstAlignmentInterval.referenceSpan.getEnd()
+                                                                         : firstAlignmentInterval.referenceSpan.getStart();
+        final int jumpLandingRefLoc = secondAlignmentInterval.forwardStrand ? secondAlignmentInterval.referenceSpan.getStart()
+                                                                            : secondAlignmentInterval.referenceSpan.getEnd();
+
+
+
+        if (firstAlignmentInterval.forwardStrand) {
+            final int alpha = firstAlignmentInterval.referenceSpan.getStart(),
+                      omega = secondAlignmentInterval.referenceSpan.getStart();
+            dupSeqRepeatUnitRefSpan = new SimpleInterval(firstAlignmentInterval.referenceSpan.getContig(),
+                                                         Math.max(alpha, omega), Math.min(jumpStartRefLoc, jumpLandingRefLoc));
+            if ( (alpha <= omega && jumpStartRefLoc < jumpLandingRefLoc) || (alpha > omega && jumpLandingRefLoc < jumpStartRefLoc) ) {
+                invertedTransInsertionRefSpan = new SimpleInterval(firstAlignmentInterval.referenceSpan.getContig(),
+                        Math.min(jumpStartRefLoc, jumpLandingRefLoc) + 1, Math.max(jumpStartRefLoc, jumpLandingRefLoc));
+            }
+            dupSeqStrandOnCtg = DEFAULT_INV_DUP_CTG_STRANDs_FR;
+        } else {
+            final int alpha = firstAlignmentInterval.referenceSpan.getEnd(),
+                      omega = secondAlignmentInterval.referenceSpan.getEnd();
+            dupSeqRepeatUnitRefSpan = new SimpleInterval(firstAlignmentInterval.referenceSpan.getContig(),
+                    Math.max(jumpStartRefLoc, jumpLandingRefLoc), Math.min(alpha, omega));
+            if ( (alpha >= omega && jumpLandingRefLoc < jumpStartRefLoc) || (alpha < omega && jumpStartRefLoc < jumpLandingRefLoc) ) {
+                invertedTransInsertionRefSpan = new SimpleInterval(firstAlignmentInterval.referenceSpan.getContig(),
+                        Math.min(jumpStartRefLoc, jumpLandingRefLoc) + 1, Math.max(jumpStartRefLoc, jumpLandingRefLoc));
+            }
+            dupSeqStrandOnCtg = DEFAULT_INV_DUP_CTG_STRANDs_RF;
+        }
+        cigarStringsForDupSeqOnCtg = DEFAULT_CIGAR_STRINGS_FOR_DUP_SEQ_ON_CTG; // not computing cigars because alt haplotypes will be extracted
+
+        dupAnnotIsFromOptimization = false;
+    }
+
+    public byte[] extractAltHaplotypeForInvDup(final ChimericAlignment chimericAlignment, final byte[] contigSeq) {
+
+        final AlignmentInterval firstAlignmentInterval  = chimericAlignment.regionWithLowerCoordOnContig;
+        final AlignmentInterval secondAlignmentInterval = chimericAlignment.regionWithHigherCoordOnContig;
+
+        final int start, end; // intended to be 0-based, semi-open [start, end)
+        final boolean needRC;
+        if (firstAlignmentInterval.forwardStrand) {
```

Comment:
Not a big deal, and maybe you feel like it might make things more confusing, but it seems like you could extract this boolean out and xor it with the booleans below and not need if/else clauses with largely duplicated code.

Reply:
Actually, the details are not duplicated (though they look very similar), so trying to go for code simplicity might, as you suspected, make the code much less readable, so I'm going to keep it the way as it is right now.
But, I do feel that we need some "contig normalization procedure" up front, which would guarantee us that the representation we get from the previous stages are restricted. But that's for future.